lib/types.d.ts

```diff
@@ -85,30 +85,70 @@ type Registration = {
      */
     preferredEmail: string;
   };
-  // TODO: Allow for qualifiers like "how old will you be at the day of the event?"
-  // TODO: Allow this to be dynamically defined by the organizers
-  // TODO: responses: { [questionId: string]: Question }
   age: number;
   gender: string;
   race: string;
   ethnicity: string;
+};
+
+type SchoolInfo = {
   university: string;
   major: string;
   studyLevel: string;
+};
+
+type ExperienceInfo = {
   hackathonExperience: number;
   softwareExperience: string;
   heardFrom: string;
+};
+
+type EventInfo = {
   size: string;
   dietary: string[];
   accomodations: string;
+};
+
+type SponsorInfo = {
   github?: string;
   linkedin?: string;
   website?: string;
   resume?: string;
   companies: Companies[];
+};
+/**
+ * Information about a specific event registration.
+ */
+type BaseRegistration = {
+  id: string;
+  /**
+   * A UNIX timestamp corresponding to when a hacker registered for the event.
+   */
+  timestamp: number;
+
+  // TODO: Allow for qualifiers like "how old will you be at the day of the event?"
+  // TODO: Allow this to be dynamically defined by the organizers
+  // TODO: responses: { [questionId: string]: Question }
+
   //claims: []; //Array of Strings will be used to id any claims (lunch, merch, etc.) made by user
+} & GeneralInfo &
+  EventInfo;
+
+type AvailabilityInfo = {
+  isFullyAvailabile: boolean;
+  timesAvailabile: string;
```

Comment:
Might need to be of another type if we plan to have HackPortal automate shift assignment.